test/Microsoft.AspNet.Mvc.FunctionalTests/Logging/LoggingExtensions.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using LoggingWebSite;
+using Microsoft.AspNet.WebUtilities;
+
+namespace Microsoft.AspNet.Mvc.FunctionalTests
+{
+    public static class LoggingExtensions
+    {
+        public const string RequestTraceIdQueryKey = "RequestTraceId";
+
+        public static IEnumerable<LogInfoDto> GetStartupLogs(this IEnumerable<ActivityContextDto> activities)
+        {
+            return activities.Where(activity => activity.RequestInfo == null)
+                             .GetLogs();
+        }
+
+        public static IEnumerable<LogInfoDto> GetStartupLogs(this IEnumerable<ActivityContextDto> activities,
+                                                            string scopeName)
+        {
+            return activities.Where(activity => activity.RequestInfo == null)
+                             .GetLogsUnderScope(scopeName);
+        }
+
+        public static IEnumerable<LogInfoDto> GetLogs(this IEnumerable<ActivityContextDto> activities,
+                                                     string requestTraceId)
+        {
+            return activities.FilterByRequestTraceId(requestTraceId)
+                             .GetLogs();
+        }
+
+        public static IEnumerable<LogInfoDto> GetLogs(this IEnumerable<ActivityContextDto> activities,
+                                                     string requestTraceId,
+                                                     string scopeName)
+        {
+            return activities.FilterByRequestTraceId(requestTraceId)
+                             .GetLogsUnderScope(scopeName);
+        }
+
+        private static IEnumerable<LogInfoDto> GetLogsUnderScope(this IEnumerable<ActivityContextDto> activities,
+                                                        string scopeName)
+        {
+            var logInfos = new List<LogInfoDto>();
+
+            foreach (var activity in activities)
+            {
+                if (activity.RepresentsScope)
+                {
+                    GetLogsUnderScopeHelper(activity.Root, logInfos, scopeName, foundScope: false);
+                }
+            }
+
+            return logInfos;
+        }
+
+        private static IEnumerable<LogInfoDto> GetLogs(this IEnumerable<ActivityContextDto> activities)
+        {
+            // Build a flat list of log messages from the log node tree 
```

Comment:
@rynowak ...I updated this PR with a new extension called `FindByScopeName` as we talked about and also added couple of tests demonstrating it...

Regarding about flattening the tree...here the current tests (startup logs) look for logs based on the logged structure's type...these logs are not part of any scope (currently) for me to use a method like `FindByScopeName` etc. and get messages out of it...so if we are not flattening the tree, we would have to however traverse through the entire(worst case scenario) tree structure to find the log message having this structure type...